persistence/src/main/java/com/epam/esm/persistence/repository/CrudRepository.java

```diff
@@ -1,48 +1,74 @@
 package com.epam.esm.persistence.repository;
 
-import com.epam.esm.domain.AbstractEntity;
+import com.epam.esm.model.AbstractEntity;
 
 import java.util.List;
 
 public interface CrudRepository<T extends AbstractEntity> {
 
```

Comment:
In deed this doesn't make much sense since method defines a contract. And by having a default implementation for all methods we loose any contract. I've reviewed all the usages of interface and came to conclusion that read methods are required anyways so I removed the default implementation for them.

Reply:
anyway default methods were added to serve compatibility issues for older versions, avoid using them in other situations

Reply:
I see now that they are used to add functionality to interfaces without breaking their older implementations. I've rewritten the interface so that it doesn't use default implementations and forbade the operations on implementation level. But I wonder if it's worth splitting the CRUD interface into C, R, U and D interfaces and implement needed ones. Intuitively it looks odd to me.